Source/Urho3D/AngelScript/Addons.cpp

```diff
@@ -1112,6 +1112,18 @@ int CScriptArray::FindByRef(asUINT startAt, void *ref) const
     return -1;
 }
 
+bool CScriptArray::Swap(CScriptArray& other)
+{
+    if (&other != this && other.GetArrayObjectType() == GetArrayObjectType())
+    {
+        SArrayBuffer* t = buffer;

```

Comment:
`Urho3D::Swap(t, buffer);` is better, I think.
Also I think that `&other != this` is redundant.